tutorials/layout/index.md

```diff
@@ -0,0 +1,1003 @@
+
+<img src="images/listview.png" style="border:1px solid black" alt="xx">
+
+Uses ListView to display a list of businesses using ListItems.
+A Divider separates the theaters from the restaurants.<br>
+**Code:** See below.
+
+</div> <div class="col-md-6" markdown="1">
+
+<img src="images/listview-flutter-gallery.png" style="border:1px solid black" alt="xx">
+
+Uses ListView to display the
+[Colors](https://docs.flutter.io/flutter/material/Colors-class.html)
+from the
+[material design palette](https://material.io/guidelines/style/color.html)
+for a particular color family.<br>
+**Code:** [colors_demo.dart](https://github.com/flutter/flutter/blob/master/examples/flutter_gallery/lib/demo/colors_demo.dart)
+from the [Flutter
+Gallery](https://github.com/flutter/flutter/tree/master/examples/flutter_gallery)
+
+</div> </div>
+
+[PENDING: Seth, I'd really like that ellipsis function!]
+
+{% include includelines filename="_code/listview/main.dart" start=31 count=105 %}
+
+{% include includelines filename="_code/listview/main.dart" start=134 count=3 %}
+
+<hr>
+
+### Stack
+
+Use [Stack](https://docs.flutter.io/flutter/widgets/Stack-class.html)
+to arrange widgets on top of a base widget&mdash;often an image.
+The widgets can completely or partially overlap the base widget.
+
+A Stack can't scroll its content.
+
+#### Stack summary:
+
+* Use for widgets that overlap another widget
+* The first widget in the list of children is the base widget
+  Subsequent children are overlaid on top of that base widget
+* A Stack's content can't scroll
+* You can choose to clip children that exceed the render box
+
+#### Stack examples:
+
+<div class="row"> <div class="col-md-6" markdown="1">
+
+<img src="images/stack.png" style="border:1px solid black" alt="xx">
+
+Uses Stack to overlay a Container (that displays its Text on a translucent
+black background) on top of a Circle Avatar.
+The Stack offsets the text using the `alignment` property and
+FractionalOffsets.<br>
+**Code:** See below.
+
+</div> <div class="col-md-6" markdown="1">
+
+<img src="images/stack-flutter-gallery.png" style="border:1px solid black" alt="xx">
+
+Uses Stack to overlay a gradient to the top of the image. The gradient
+ensures that the toolbar's icons are distinct against the image.<br>
+**Code:** [contacts_demo.dart](https://github.com/flutter/flutter/blob/master/examples/flutter_gallery/lib/demo/contacts_demo.dart)
+from the [Flutter
+Gallery](https://github.com/flutter/flutter/tree/master/examples/flutter_gallery)
+
+</div> </div>
+
+{% include includelines filename="_code/stack/main.dart" start=39 count=21 %}
+
+<hr>
+
+### Card
+
+A Card, from the material library, contains related nuggets of information
+and can be composed from most any widget, but is often used with ListItem.
+Card has a single child, but its child can be a column, row, list, grid,
+or other widget that supports multiple children. By default, Card expands
+to fill available space. You can use
```

Comment:
Card doesn't expand by default.  By default, it shrinks down to 0x0:

```dart
void main() => runApp(new Center(child: new Card(color: const Color(0xFF00FF00))));
```

Notice that there's no green.